domain/machine/service/machine_cloud_instance.go

```diff
@@ -27,23 +30,21 @@ func (s *Service) GetInstanceID(ctx context.Context, machineUUID machine.UUID) (
 	return instance.Id(instanceId), nil
 }
 
-// GetInstanceIDByMachineName returns the cloud specific instance id for this machine.
-// The following errors may be returned:
-//   - [github.com/juju/juju/domain/machine/errors.NotProvisioned] if the machine
-//     is not provisioned.
-func (s *Service) GetInstanceIDByMachineName(ctx context.Context, machineName machine.Name) (instance.Id, error) {
+// GetPollingInfos returns the polling information for the specified machines.
+func (s *Service) GetPollingInfos(ctx context.Context, machineNames []machine.Name) (domainmachine.PollingInfos,
+	error) {
 	ctx, span := trace.Start(ctx, trace.NameFromFunc())
 	defer span.End()
 
-	machineUUID, err := s.st.GetMachineUUID(ctx, machineName)
-	if err != nil {
-		return "", errors.Errorf("retrieving machine UUID for machine %q: %w", machineName, err)
+	validationError := errors.Join(transform.Slice(machineNames, machine.Name.Validate)...)
```

Comment:
Nice: I didn't realize you could use the `Slice` method to run an existing method of the slice type. 👍